bus-mapping/src/circuit_input_builder.rs

```diff
@@ -255,8 +318,22 @@ impl<'a, C: CircuitsParams> CircuitInputBuilder<C> {
                 &geth_trace.struct_logs[index..],
             )?;
             tx.steps_mut().extend(exec_steps);
-        }
 
+            // Generate EndChunk and proceed to the next if it's not the last chunk
+            if is_chunked && self.chunk_ctx.rwc.0 >= self.circuits_params.max_rws() {
+                assert!(self.chunk_ctx.is_last_chunk(), "Fixed max rws is ");
+                // Update param accordding to number of rws actually generated
+                // the initial self.circuits_params function as a chunking thresholded but not
+                // constrain
+                if self.chunk_ctx.is_dynamic {
+                    self.cur_chunk_mut().fixed_param = self.compute_param(&self.block.eth_block);
+                }
+
+                self.set_end_chunk(tx.last_step());
+                self.commit_chunk(true);
+                self.set_begin_chunk(tx.last_step());
```

Comment:
Here I think we should pass `next_tx.first_step()` to both `set_end_chunk` and `set_begin_chunk` instead of `tx.last_step()`.

Reply:
Any reasons why?  Shouldn't these two be the same? Also `tx.last_step()` returns the last ExecStep from this extension `tx.steps_mut().extend(exec_steps)` after we gen_associated_ops for the current GethExecStep, which means we don't know what the next tx first step is before processing the next tx

Reply:
> Shouldn't these two be the same?

Yes, both `set_end_chunk(STEP)` and `set_begin_chink(STEP)` should put same `STEP`

However, an execution step keep the state **BEFORE** execution this step.
Thus in end_chunk slot what we need to put is the **new state**, not the old state. 
new state is kept in next_step pre-state, that's why we need to put next step instead of `last step`

Reply:
I left another reply https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1690/files#r1436479749 major regarding to overall design. Cite here since related